### PR TITLE
Clearer feature flags

### LINKED ISSUE #374: TGA image example code in documentation doesn't compile
- Version of embedded-graphics in use: 0.6.2
- Version of tinytga in use: 0.3.2
- Target: simulator

## Description of the problem

This example code doesn't seem to compile for me: https://docs.rs/embedded-graphics/0.6.2/embedded_graphics/#display-a-tga-image . I get the following error:
```
error[E0308]: mismatched types
  --> src/main.rs:79:34
   |
79 |             let img = Image::new(&tga, Point::zero());
   |                                  ^^^^ expected struct `embedded_graphics::image::image_raw::ImageRaw`, found struct `tinytga::Tga`
   |
   = note: expected reference `&embedded_graphics::image::image_raw::ImageRaw<'_, _, _>`
              found reference `&tinytga::Tga<'_>`
```

### LINKED ISSUE #375: Improve documentation of graphics flag for tinytga and tinybmp
The documentation for the `graphics` flag in `tinytga` and `tinybmp` is hard to discover which recently caused problems for at least two users. We should add a notes about the required feature to the examples in e-g (like suggested in #374), the `tinybmp` and `tinytga` readme files and the simulator examples.

### PR DESCRIPTION
**Pull request by jamwaffles**
Hi! Thank you for helping out with Embedded Graphics development! Please:

- [x] Check that you've added passing tests and documentation
- [x] Add a simulator example(s) where applicable
- [x] Add a `CHANGELOG.md` entry in the **Unreleased** section under the appropriate heading (**Added**, **Fixed**, etc) and appropriate crate (`embedded-graphics`, `simulator`, `tinytga`, `tinybmp`) if your changes affect the **public API**
- [x] Run `rustfmt` on the project
- [x] Run `./build.sh` (Linux/macOS only) and make sure it passes. If you use Windows, check that CI passes once you've opened the PR.

## PR description

Adds notes and comments about the `graphics` feature of tinytga and tinybmp.

This also adds in the sector and arc example generators - it looks like the example was added manually in #310 but not to the generator.

Closes #374 
Closes #375

### CODE REVIEW

**Comment by jamwaffles** [comment]
There might be some missing comments/notes in various `README.md`s, but I'd like to fix those with #381 later.

File: tinytga/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,80 @@
-//! No-std compatible TGA parser designed for embedded systems, but usable anywhere
+//! A small TGA parser designed for embedded, no-std environments but usable anywhere. Beyond
+//! parsing the image header, no other allocations are made. A reference to the input image data is
+//! kept and slices are returned from it.
+//!
+//! Call `Tga.into_iter()` to get an iterator over individual pixels in the image.
```
Comment (rfuest) [inline tinytga/src/lib.rs:5]: This should also be added to the `tinybmp` docs.

IMO `Tga.into_iter()` could be confused with `Tga::into_iter()`.
```suggestion
//! To access the individual pixels in the images the `Tga` struct implements `IntoIterator`.
//! The example below demonstrates this feature by collect all pixels in a vector.
```

**Comment by jamwaffles** [comment]
Good points, thanks. Changes addressed!

File: tinytga/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,82 @@
-//! No-std compatible TGA parser designed for embedded systems, but usable anywhere
+//! A small TGA parser designed for embedded, no-std environments but usable anywhere. Beyond
+//! parsing the image header, no other allocations are made.
+//!
+//! To access the individual pixels in an image, the [`Tga`] struct implements `IntoIterator`. It is
+//! also possible to access the raw image data by reading the [`pixel_data`] field.
```
Comment (rfuest) [inline tinytga/src/lib.rs:5]: I think this doesn't make it clear enough that `pixel_data` might return data that is still RLE encoded.

File: embedded-graphics/src/image/mod.rs [hunk 3]
```diff
@@ -15,6 +15,9 @@
 //! This example loads a TGA-formatted image using the [tinytga] crate and draws it to the display
 //! using the [`Image`] wrapper. The image is positioned at the top left corner of the display.
 //!
+//! The `graphics` feature of `tinytga` needs to be enabled in `Cargo.toml` to use the `Tga` object
+//! with embedded-graphics.
```
Comment (rfuest) [inline embedded-graphics/src/image/mod.rs:19]: I was lazy when I created the last review and forgot to mention that this suggestion was meant to also apply to the other instances that describe the need for the `graphics` feature. IMO it's especially important to change this in `lib.rs`.

File: tinytga/src/lib.rs (hunk 2 above)
Comment (jamwaffles) [reply tinytga/src/lib.rs:5]: I'm not sure how to make "raw" clearer. "raw image **file** data"? "raw, **possibly still compressed** image data"?

File: embedded-graphics/src/image/mod.rs (hunk 3 above)
Comment (jamwaffles) [reply embedded-graphics/src/image/mod.rs:19]: No problem, I've updated the other places too. Tbf I was lazy when I copypasta'd this comment throughout the codebase.

File: tinytga/src/lib.rs [hunk 4]
```diff
@@ -1,4 +1,83 @@
...
```
Comment (rfuest) [inline tinytga/src/lib.rs:5]:
```suggestion
//! To access the individual pixels in an image, the [`Tga`] struct implements `IntoIterator`. It is
//! also possible to access the unaltered raw image data by reading the [`pixel_data`] field. This data will need
//! to be interpreted according to the [`image_type`] specified in the header.
```

**Comment by jamwaffles** [comment]
Thanks for the suggestion. Ready for another look

p.s. is it enough to just click the "rerequest review" button, or would you prefer a comment as well? I think review re-requests send an email, right?

**Comment by rfuest** [comment]
> p.s. is it enough to just click the "rerequest review" button, or would you prefer a comment as well? I think review re-requests send an email, right?

Yes, the re-requests also send a notification email. If there are no additional comments you want to make about the changes it's enough to press the button.

**Comment by jamwaffles** [comment]
Thanks!

> If there are no additional comments you want to make about the changes it's enough to press the button.

Noted, I'll reduce the spam from now on :)

**Pull request merged**